api/cluster.go

```diff
@@ -546,6 +549,19 @@ func (s *Server) peerQuerySpeculativeChan(ctx context.Context, data cluster.Trac
 				}
 
 				if resp.err != nil {
+					// check if there is another peer for this shardGroup. If so try it.
```

Comment:
The reason I originally didn't retry errors, is that many errors aren't effective to retry. Things like bad requests will still fail with the peers. I would be interested to see how often this is actually helpful.

Reply:
Between when cluster.MembersForSpeculativeQuery() returned the list of peers and when the requests where actually run, nodes could go offline.  We have been seeing bursts of failures in clusters that handle a high volume of requests because of this.

Reply:
no instance should issue a request that is considered a bad request by a peer. if we see that, it is a bug to be fixed.

Reply:
discussion on this topic lives in #985